client/client.go

```diff
@@ -275,6 +284,7 @@ func (c *DatabricksClient) perform(
 	visitors = append([]func(*http.Request) error{
 		c.Config.Authenticate,
 		c.addHostToRequestUrl,
+		c.addApplicationJsonContentType,
```

Comment:
I know this doesn't change behavior, but strictly speaking we only need to set this header on requests with a body. We can set the `Accept` header on all of them as it pertains to the response.

Reply:
we set the multipart header or octet-stream header in request visitors. multipart requests do require a Content-Type because of the multipart boundary.

Reply:
https://github.com/databricks/databricks-sdk-go/blob/upload-download/service/workspace/utilities.go#L171-L175